presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueries.java

```diff
@@ -392,6 +392,98 @@ private Session enableDecimalParsing()
                 .build();
     }
 
+    @Test
+    public void testDecimalArithmetic()
+    {
+        // Addition of two Long Decimal columns with inferred types DECIMAL(35,20) and DECIMAL(29,4)
+        // and also contains NULLs.
+        assertQuery(
+                "SELECT n + m from (values " +
+                        "(DECIMAL'999999999999999.999' , DECIMAL'1')," +
+                        "(DECIMAL'-123456789012345.123456', DECIMAL'-9999999999999999')," +
+                        "(DECIMAL'1.23', DECIMAL'-0.0005')," +
+                        "(DECIMAL'1.33333333333333333333', DECIMAL'-0.0005')," +
+                        "(NULL, NULL), (decimal'1.23', NULL)) t(n, m)");
```

Comment:
Hi Karteek, Queries that process only literal values like this may be constant folded by Prestissimo coordinator without invoking Velox evaluation. Could you try selecting from columns instead of literal values only? (cc @amitkdutta)

Reply:
I understand just executing literals may be constant folded. But, here we are testing arithmetic operation and Presto Query plan has SUBTRACT operator here:
<img width="702" alt="Screen Shot 2022-11-30 at 11 16 55 AM" src="https://user-images.githubusercontent.com/6971561/204888702-263e5603-8f8a-4724-b027-078972f5e29e.png">

Reply:
My concern is that the constant folding my use Presto's code instead of Velox code (that you would like to test IIUC). Let me try if I can verify this locally.

Reply:
@kagamiori lmk if you have any other questions.

Reply:
The concern is valid. From what I understand if the plan says there is a SUBTRACT operator, then that is what gets executed in the backend. Or else, we would just see a project node. CMIW.

Reply:
@kagamiori did you get a chance to verify this?

Reply:
Hi @karteekmurthys, I'm having trouble running the e2e query on my laptop (due to my setup of Prestissimo). I'm asking @amitkdutta and @mbasmanova to help confirm.

Reply:
@karteekmurthys Karteek, to speed up the review, would you share the output of EXPLAIN (TYPE DISTRIBUTED) command for one of these queries?

Reply:
```
presto:default> explain (type distributed) SELECT n - m from (values (CAST('999999999999999.999' as decimal(18,3)), CAST('1' as decimal(1,0)))) t(n, m);
                                         Query Plan                                         
--------------------------------------------------------------------------------------------
 Fragment 0 [SINGLE]                                                                       +
     Output layout: [expr_6]                                                               +
     Output partitioning: SINGLE []                                                        +
     Stage Execution Strategy: UNGROUPED_EXECUTION                                         +
     - Output[_col0] => [expr_6:decimal(19,3)]                                             +
             Estimates: {rows: 1 (17B), cpu: 35.00, memory: 0.00, network: 0.00}           +
             _col0 := expr_6 (1:37)                                                        +
         - Project[projectLocality = LOCAL] => [expr_6:decimal(19,3)]                      +
                 Estimates: {rows: 1 (17B), cpu: 35.00, memory: 0.00, network: 0.00}       +
                 expr_6 := (field) - (field_0) (1:48)                                      +
             - LocalExchange[ROUND_ROBIN] () => [field:decimal(18,3), field_0:decimal(1,0)]+
                     Estimates: {rows: 1 (18B), cpu: 18.00, memory: 0.00, network: 0.00}   +
                 - Values => [field:decimal(18,3), field_0:decimal(1,0)]                   +
                         Estimates: {rows: 1 (18B), cpu: 0.00, memory: 0.00, network: 0.00}+
                         (DECIMAL'999999999999999.999', DECIMAL'1')                        +
                                                                                           +
                                                                                            
```

Reply:
@karteekmurthys Thanks. This plan shows that the computation (a - b) will happen on the worker.